### PR TITLE
Add payjoin take2

### PR DESCRIPTION
**Pull request by Overtorment**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ebyxttj5etvea3v7hy36u2246g

[download apk](http://filestorage.bluewallet.io:1488/1597341843.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qzt311jvju9ky7gc4kamqh2qyc

[download apk](http://filestorage.bluewallet.io:1488/1597429909.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/j5w777mg0tkdkmeeveq8rug7g0

[download apk](http://filestorage.bluewallet.io:1488/1599856004.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/amgtbqtcb590cbe25ajvuja0tr

[download apk](http://filestorage.bluewallet.io:1488/1599996749.apk)

File: screen/send/confirm.js [hunk 1]
```diff
@@ -168,11 +195,13 @@ export default class Confirm extends Component {
                 {currency.satoshiToLocalCurrency(this.state.feeSatoshi)})
               </Text>
               <BlueSpacing40 />
-              {this.state.isLoading ? (
-                <ActivityIndicator />
-              ) : (
-                <BlueButton onPress={() => this.broadcast()} title={loc.send.confirm_sendNow} />
+              {!!this.state.payjoinUrl && (
+                <View style={styles.payjoinWrapper}>
+                  <Text style={styles.payjoinText}>Payjoin</Text>
```
Comment (limpbrains) [inline screen/send/confirm.js:200]: should we translate "Payjoin", also some short description of what is Payjoin would be nice

File: screen/send/confirm.js [hunk 2]
```diff
@@ -50,59 +55,63 @@ export default class Confirm extends Component {
...
+            SegwitBech32Wallet.type,
+          ].includes(this.state.fromWallet.type)
+        ) {
+          amount = formatBalanceWithoutSuffix(amount, BitcoinUnit.BTC, false);
         }
+
+        this.props.navigation.navigate('Success', {
+          fee: Number(this.state.fee),
+          amount,
+          dismissModal: () => this.props.navigation.dismiss(),
```
Comment (limpbrains) [inline screen/send/confirm.js:111]: navigation.dismiss() was dropped in react-navigation v5.  How does it works?

File: class/payjoin-transaction.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      if (!result) {
+        throw new Error(`Broadcast failed`);
+      }
+      return '';
+    } catch (e) {
+      return 'Error: ' + e.message;
+    }
+  }
+
+  async scheduleBroadcastTx(txHex, milliseconds) {
```
Comment (limpbrains) [inline class/payjoin-transaction.js:69]: What is `scheduleBroadcastTx` for? I can't find a place where it has been used

File: screen/send/confirm.js [hunk 4]
```diff
@@ -287,11 +316,20 @@ const styles = StyleSheet.create({
...
     alignSelf: 'center',
   },
+  payjoinWrapper: {
+    flexDirection: 'row',
+    marginHorizontal: 20,
+    marginBottom: 10,
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  payjoinText: { color: '#81868e', fontSize: 14 },
```
Comment (limpbrains) [inline screen/send/confirm.js:326]: What about Dark mode, does this color fit?

File: screen/send/details.js [hunk 5]
```diff
@@ -425,9 +426,12 @@ export default class SendDetails extends Component {
         if ('label' in parsedBitcoinUri.options) {
           memo = parsedBitcoinUri.options.label || memo;
         }
+        if (Object.prototype.hasOwnProperty.call(parsedBitcoinUri.options, 'pj')) {
```
Comment (limpbrains) [inline screen/send/details.js:429]: Why so complicated? And not `'pj' in parsedBitcoinUri.options`? Probably we need tests for decodeBitcoinUri

**Review by limpbrains — CHANGES_REQUESTED** [review]

File: screen/send/confirm.js (hunk 1 above)
Comment (Overtorment) [reply screen/send/confirm.js:200]: kinda wide-used term atm, maybe we will figure out how to translate it later

File: class/payjoin-transaction.js (hunk 3 above)
Comment (Overtorment) [reply class/payjoin-transaction.js:69]: `scheduleBroadcastTx` is called from within payjoin-client, as far as I remember

File: screen/send/confirm.js (hunk 4 above)
Comment (Overtorment) [reply screen/send/confirm.js:326]: @marcosrdz what do we need to do to make this support darkmode?

File: screen/send/details.js (hunk 5 above)
Comment (Overtorment) [reply screen/send/details.js:429]: valid point. its a leftover from original contributor

File: screen/send/confirm.js (hunk 4 above)
Comment (marcosrdz) [reply screen/send/confirm.js:326]: I write screens to use hooks. useTheme automatically switches.

Comment (marcosrdz) [reply screen/send/confirm.js:326]: BlueCurrentTheme is used for class components

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8yc80w5m3nvbu4dt195g2ffd20

[download apk](http://filestorage.bluewallet.io:1488/1600341540.apk)

**Comment by Overtorment** [comment]
fyi, payjoin can be teted here: https://btc.donate.kukks.org

**Comment by Overtorment** [comment]
ok I think its ready for tests @ncoelho @marcosrdz

**Comment by ncoelho** [comment]
Oh boy! :)

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zxu9e6z93drmpbd54hpp4pu2p8

[download apk](http://filestorage.bluewallet.io:1488/1600557773.apk)

**Review by marcosrdz — APPROVED** [review]

**Comment by marcosrdz** [comment]
> ok I think its ready for tests @ncoelho @marcosrdz 

Should it be enabled by default? Or leave it as is?

**Comment by marcosrdz** [comment]
> > ok I think its ready for tests @ncoelho @marcosrdz 
> 
> Should it be enabled by default? Or leave it as is?

Maybe have it enabled by default and have the toggle in an "advanced" 3 dot menu option on confirm screen

**Comment by ncoelho** [comment]
> > ok I think its ready for tests @ncoelho @marcosrdz 
> 
> Should it be enabled by default? Or leave it as is?

I believe we discussed this on the original thread

**Comment by ncoelho** [comment]
Tested & ACK

**Review by ncoelho — APPROVED** [review]

**Comment by junderw** [comment]
Reminder: payjoin-client is not BIP compliant yet.

**Pull request merged**